src/util/math.h

```diff
@@ -45,14 +45,18 @@ constexpr bool even(T value) {
 #pragma intrinsic(fabs)
 #endif
 
-constexpr unsigned int roundUpToPowerOf2(int v) {
-    DEBUG_ASSERT(v >= 0);
-#ifdef __cpp_lib_bitops
+constexpr int roundUpToPowerOf2(int v) {
```

Comment:
Can we still make this unsigned? Using `int` for a function thats only defined for an `unsigned int`-range of values does not make sense IMO.

Reply:
My original intention was to keep the scope of the change limited and do not alter the function signature.

We have a problem anyway. It is the useless range of 0x80000000 to 0xffffffff which is either insane big positive or negative. The mendatory cast to int32_t in the using code treats it as negative anyway.

However I can imagine to turn the input and output parameter to unsigned. Than we move the overflow issue into the using code which we may ignore because it is possible but unrealistic case of failure there. 

What do you think?

Reply:
> However I can imagine to turn the input and output parameter to unsigned. Than we move the overflow issue into the using code which we may ignore because it is possible but unrealistic case of failure there.

Yes. Thats also how `std::bit_ceil` works. If you pass it the largest value, it will overflow and return 0. IMO it makes sense to just return 0 because thats not a possible return value anyways. So lets just document that a return value of 0 means failure and move on. The implementation in my original PR already did that.